beacon-chain/state/state-native/getters_withdrawal_test.go

```diff
@@ -1,597 +1,352 @@
+				s := state_native.EmptyStateFromVersion(t, stateVersion)
+				vals := make([]*ethpb.Validator, 100)
+				balances := make([]uint64, 100)
+				for i := range vals {
+					balances[i] = params.BeaconConfig().MaxEffectiveBalance
+					val := &ethpb.Validator{
+						WithdrawalCredentials: make([]byte, 32),
+						EffectiveBalance:      params.BeaconConfig().MaxEffectiveBalance,
+						WithdrawableEpoch:     primitives.Epoch(1),
+					}
+					val.WithdrawalCredentials[0] = params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
+					val.WithdrawalCredentials[31] = byte(i)
+					vals[i] = val
+				}
+				balances[3] += params.BeaconConfig().MinDepositAmount
+				balances[10] += params.BeaconConfig().MinDepositAmount
+				require.NoError(t, s.SetValidators(vals))
+				require.NoError(t, s.SetBalances(balances))
+				saved := params.BeaconConfig().MaxValidatorsPerWithdrawalsSweep
+				params.BeaconConfig().MaxValidatorsPerWithdrawalsSweep = 10
+				expected, _, err := s.ExpectedWithdrawals()
+				require.NoError(t, err)
+				require.Equal(t, 1, len(expected))
+				withdrawal := &enginev1.Withdrawal{
+					Index:          0,
+					ValidatorIndex: 3,
+					Address:        vals[3].WithdrawalCredentials[12:],
+					Amount:         params.BeaconConfig().MinDepositAmount,
+				}
+				require.DeepEqual(t, withdrawal, expected[0])
+				params.BeaconConfig().MaxValidatorsPerWithdrawalsSweep = saved
+			})
+		})
+	}
+
+	t.Run("electra all pending partial withdrawals", func(t *testing.T) {
+		// Load a serialized Electra state from disk.
+		// This spectest has a fully hydrated beacon state with partial pending withdrawals.
+		serializedBytes, err := util.BazelFileBytes("tests/mainnet/electra/operations/execution_layer_withdrawal_request/pyspec_tests/pending_withdrawals_consume_all_excess_balance/pre.ssz_snappy")
 		require.NoError(t, err)
-		require.Equal(t, params.BeaconConfig().MaxWithdrawalsPerPayload, uint64(len(expected)))
-		withdrawal := &enginev1.Withdrawal{
-			Index:          0,
-			ValidatorIndex: 0,
-			Address:        s.validators[0].WithdrawalCredentials[12:],
-			Amount:         params.BeaconConfig().MaxEffectiveBalance,
-		}
-		require.DeepEqual(t, withdrawal, expected[0])
-	})
-	t.Run("all fully and partially withdrawable", func(t *testing.T) {
-		s := BeaconState{
-			version:    version.Deneb,
-			validators: make([]*ethpb.Validator, 100),
-			balances:   make([]uint64, 100),
-		}
-		for i := range s.validators {
-			s.balances[i] = params.BeaconConfig().MaxEffectiveBalance + 1
-			val := &ethpb.Validator{
-				WithdrawalCredentials: make([]byte, 32),
-				EffectiveBalance:      params.BeaconConfig().MaxEffectiveBalance,
-				WithdrawableEpoch:     primitives.Epoch(0),
-			}
-			val.WithdrawalCredentials[0] = params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
-			s.validators[i] = val
-		}
-		expected, err := s.ExpectedWithdrawals()
+		serializedSSZ, err := snappy.Decode(nil /* dst */, serializedBytes)
```

Comment:
This could be a nice general test helper (get example state for fork).

Reply:
If someone finds that useful, I would be happy to help move this to some shared method. I thought this was a rather specific example that I crawled the spectests repo to find.